lang_php/analyze/php_to_generic.ml

```diff
@@ -263,11 +265,16 @@ and expr =
   | Obj_get (v1, t, Id [v2]) ->
       let v1 = expr v1 and v2 = ident v2 in
       G.DotAccess (v1, t, G.EId (v2, G.empty_id_info()))
+  | Obj_get (v1, t, Metavar v2) ->
```

Comment:
ok, got it. This is the main issue. But again an alternative is to add a case like
| Obj_get (v1, t, Var ((s, _t)) when AST_generic_helpers.is_metavar_name s && !Flag_parsing.sgrep_mode -> 
 ...

It's just that in sgrep_mode, we really want to treat those Var like Id, which we do when converting to the generic AST, but the problem is that each time we do deeper match where we use Id, then we need also to catch Var (s,_) when is_metavar_name && sgrep_mode.